bundles/org.smarthomej.binding.notificationsforfiretv/src/main/java/org/smarthomej/binding/notificationsforfiretv/internal/NotificationsForFireTVConnection.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ * Copyright (c) 2021 Contributors to the SmartHome/J project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.smarthomej.binding.notificationsforfiretv.internal;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Tom Blum - Initial contribution
+ */
+@NonNullByDefault
+public class NotificationsForFireTVConnection {
+
+    private final Logger logger = LoggerFactory.getLogger(NotificationsForFireTVConnection.class);
+
+    private String boundary;
+    private static final String LINE = "\r\n";
+    private static final String PROTOCOL = "http";
+    private HttpURLConnection httpConn;
+    private OutputStream outputStream;
+    private PrintWriter writer;
+
+    /**
+     * This constructor initializes a new HTTP POST request with content
+     * type is set to multipart/form-data
+     *
+     * @param ip
+     * @param port
+     * @throws IOException
+     */
+    public NotificationsForFireTVConnection(String ip, int port) throws IOException {
+        boundary = UUID.randomUUID().toString();
```

Comment:
This looks very difficult. Did you have a look at Jetty's MultipartContentProvider, e.g. https://stackoverflow.com/a/38984173? Since we already have Jetty aboard, it looks much easier to use that than construct the MultiPart thing on your own.

Reply:
Tried, and for me it makes it more complicated. The example does not show how to create a client instance. If i use HttpClient it does not have a newRequest method. I tried to use jetty in the past and there where a lot of issues about for example webdav, propfind, aso methods. It looks easy to use but at the end its more complicated. So i prefer to use the same way like we did on the AEC-Binding, so to use HttpURLConnection.

Reply:
Ok. I‘m fine with that, too. Maybe it‘s because I used jetty in the http binding and found it very easy.

Reply:
BTW "Notifications for Fire TV" is very picky about headers. As i remember jetty adds some default headers which maybe the server is not able to handle. So i don't know if this will also produces more problems.

Reply:
keep it